src/memcached-promisify.js

```diff
@@ -0,0 +1,120 @@
+// forked from https://github.com/flightstats/memcached-promisify
+
+import { assign } from 'lodash';
+import Memcached from 'memcached';
+
+const MAX_EXPIRATION = 60 * 60 * 24 * 30; // memcached's max exp in seconds (30 days)
+
+function cachePromise(client, method, ...args) {
+    return new Promise((resolve, reject) => {
+        client._cache[method](...args, (err, data) => { // eslint-disable-line no-underscore-dangle
+            if (err) {
+                reject(err);
+            } else {
+                resolve(data);
+            }
+        });
+    });
+}
+
+/**
+ * Cache
+ * @constructor
+ * @param {Object} config - cache key prefix
+ * @param {string} [config.keyPrefix] - cache key prefix
+ * @param {string} [config.cacheHost] - cache host url
+ * @param {Object} [options] - options passed to memcached
+ *
+ * @example
+ *    new Cache('prefix', 'host');
+ *
+ */
+class Cache {
+    constructor(config, options = { maxExpiration: MAX_EXPIRATION }) {
+        this.config = assign({
+            keyPrefix: '',
+            cacheHost: 'localhost:11211',
+        }, config);
+
+        this.config.maxExpiration = options.maxExpiration;
+        console.log('creating memcached instance for cache host:', this.config.cacheHost); // eslint-disable-line
+        // eslint-disable-next-line no-underscore-dangle
+        this._cache = new Memcached(this.config.cacheHost, options);
+    }
+
+    /**
+    * get a cache item
+    * @param {string} key - cache key
+    * @returns {Promise}
+    */
+    get(key) {
+        return cachePromise(this, 'get', `${this.config.keyPrefix}${key}`);
+    }
+
+    /**
+     * get a cache item
+     * @param {string} key - full prefixed cache key
+     * @returns {Promise}
+     */
+    utilGet(key) {
+        return cachePromise(this, 'get', key);
+    }
+
+    /**
+     * get an object of cache items
+     * @param {array} keys - an array of cache keys
+     * @returns {Promise}
```

Comment:
Can we have more details about the result. Is it an array? Will it return null if it's going to fail, or will it raise an exception.
The same is true to other functions